mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -214,7 +216,7 @@ private[ml] object TreeClassifierParams {
 /**
  * Parameters for Decision Tree-based regression algorithms.
  */
-private[ml] trait TreeRegressorParams extends Params {
+private[ml] trait TreeRegressorParams extends DecisionTreeParams with HasVarianceCol {
```

Comment:
I think the best solution here would be to create a DecisionTreeRegressorParams trait which inherits from the other param traits.  It can implement validateAndTransformSchema.  The estimator & model can inherit from it to share the implementation.